### DISCUSSION TITLE
Linker Error. fastrtps 2.13.1 expects libfastcdr.so.1 as a dependency

### DISCUSSION THREAD

**Discussion by bshantam97** [eProsima/Fast-DDS#4379] [Support]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

The expected behaviour should be that all the cdr functions should be available to fastrtps and there should be no linker error.

### Current behavior

The current behaviour gives the following error 
/usr/bin/ld: warning: libfastcdr.so.1, needed by /usr/local/lib/libfastrtps.so.2.13.1, not found (try using -rpath or -rpath-link)
/usr/bin/ld: /usr/local/lib/libfastrtps.so.2.13.1: undefined reference to `eprosima::fastcdr::Cdr::deserializeArray(int*, unsigned long)'
/usr/bin/ld: /usr/local/lib/libfastrtps.so.2.13.1: undefined reference to `eprosima::fastcdr::Cdr::deserializeArray(char*, unsigned long)'
/usr/bin/ld: /usr/local/lib/libfastrtps.so.2.13.1: undefined reference to `eprosima::fastcdr::Cdr::deserializeArray(short*, unsigned long)'
/usr/bin/ld: /usr/local/lib/libfastrtps.so.2.13.1: undefined reference to `eprosima::fastcdr::Cdr::setState(eprosima::fastcdr::Cdr::state&)'
/usr/bin/ld: /usr/local/lib/libfastrtps.so.2.13.1: undefined reference to `eprosima::fastcdr::Cdr::serializeArray(int const*, unsigned long)'
/usr/bin/ld: /usr/local/lib/libfastrtps.so.2.13.1: undefined reference to `eprosima::fastcdr::Cdr::Cdr(eprosima::fastcdr::FastBuffer&, eprosima::fastcdr::Cdr::Endianness, eprosima::fastcdr::Cdr::CdrType)'
/usr/bin/ld: /usr/local/lib/libfastrtps.so.2.13.1: undefined reference to `eprosima::fastcdr::Cdr::readWString[abi:cxx11](unsigned int&)'
/usr/bin/ld: /usr/local/lib/libfastrtps.so.2.13.1: undefined reference to `eprosima::fastcdr::Cdr::readString(unsigned int&)'
/usr/bin/ld: /usr/local/lib/libfastrtps.so.2.13.1: undefined reference to `eprosima::fastcdr::Cdr::serializeArray(char const*, unsigned long)'
/usr/bin/ld: /usr/local/lib/libfastrtps.so.2.13.1: undefined reference to `eprosima::fastcdr::Cdr::serializeArray(short const*, unsigned long)'

### Steps to reproduce

For fastdds 
git clone --recursive https://github.com/eProsima/Fast-DDS.git
git checkout v2.13.1

For fastCDR
git clone --recursive https://github.com/eProsima/Fast-CDR.git
git checkout v2.1.3

Build both these libraries and then write a custom publisher and subscriber to run fastDDS.

### Fast DDS version/commit

FastDDS: v2.13.1
FastCDR: v2.1.3

### Platform/Architecture

Other. Please specify in Additional context section.

### Transport layer

Default configuration, UDPv4 & SHM

**Comment by Mario-DL** [comment]
Hi @bshantam97 

Thanks for using Fast DDS. I could not reproduce the issue since it is most likely related with the build configuration of your application. If you are using `cmake` ensure that you are correctly linking against `fastrtps`, `fastcdr` and `foonathan_memory` with `target_link_libraries()` apart from having `asio` and `tinyxml2` in your system.

In addition, ensure that the types are regenerated with the latest version of `Fast DDS Gen`
 
I am moving this to the proper support forum.

**Reply by bshantam97** [reply]
@Mario-DL Thank you for your response. I was able to solve this issue by setting the -Dfastcdr_DIR and -Dfoonathan_memory_DIR paths while configuring cmake. The full command I used was cmake ../ -Dfastcdr_DIR=path/to/fastcdr cmake configs -Dfastcdr_DIR=/path/to/foonathan_memory/cmake configs.
I had ROS2 installed so by default fastDDS was taking the default fastCDR installation in ROS2. 
What I am thinking is that since fastDDS is backward compatible with fastCDR V1 since v2.12.1 the linker error technically should not have happened.
